include/z64save.h

```diff
@@ -24,6 +24,36 @@ typedef enum RespawnMode {
 
 #define SAVE_BUFFER_SIZE 0x4000
 
+typedef enum {
+    /* 0  */ MAGIC_STATE_IDLE, // Regular gameplay
+    /* 1  */ MAGIC_STATE_CONSUME_SETUP, // Sets the speed at which the magic border flashes
+    /* 2  */ MAGIC_STATE_CONSUME, // Consume magic until target is reached or no more magic is available
+    /* 3  */ MAGIC_STATE_METER_FLASH_1, // Flashes border
+    /* 4  */ MAGIC_STATE_METER_FLASH_2, // Flashes border and draws yellow magic to preview target consumption
+    /* 5  */ MAGIC_STATE_RESET, // Reset colors and return to idle
+    /* 6  */ MAGIC_STATE_METER_FLASH_3, // Flashes border with no additional behaviour
+    /* 7  */ MAGIC_STATE_CONSUME_LENS, // Magic slowly consumed by lens
+    /* 8  */ MAGIC_STATE_STEP_CAPACITY, // Step `magicCapacity` to full capacity
+    /* 9  */ MAGIC_STATE_FILL, // Add magic until magicFillTarget is reached
+    /* 10 */ MAGIC_STATE_CONSUME_GORON_ZORA_SETUP,
+    /* 11 */ MAGIC_STATE_CONSUME_GORON_ZORA, // Magic slowly consumed by goron spikes or zora shocks
+    /* 12 */ MAGIC_STATE_CONSUME_GIANTS_MASK // Magic slowly consumed by giants mask
+} MagicState;
+
+typedef enum {
+    /* 0 */ MAGIC_CONSUME_NOW, // Consume Magic immediately without preview
+    /* 1 */ MAGIC_CONSUME_WAIT_NO_PREVIEW, // Sets consume target but waits to consume. No yellow magic preview to target consumption. Unused
+    /* 2 */ MAGIC_CONSUME_NOW_ALT, // Identical behaviour to MAGIC_CONSUME_NOW. Unused
+    /* 3 */ MAGIC_CONSUME_LENS, // Lens consumption
+    /* 4 */ MAGIC_CONSUME_WAIT_PREVIEW, // Sets consume target but waits to consume. Draws yellow magic to target consumption
+    /* 5 */ MAGIC_CONSUME_GORON_ZORA, // Zora shock and Goron Spike Roll slow consumption
```

Comment:
```suggestion
    /* 5 */ MAGIC_CONSUME_GORON_ZORA, // Zora electric barrier and Goron spiked rolling slow consumption
```
Or at least make the terminology consistent.

Reply:
Ah. I like barrier. I'll stick with that